src/Cache/Signature.php

```diff
@@ -73,4 +73,19 @@ public function equals(SignatureInterface $signature)
 
         return true;
     }
+
+    private static function utf8Encode(array $data)
+    {
+        if (!function_exists('utf8_encode') || !function_exists('mb_detect_encoding')) {
+            return $data;
+        }
+
+        array_walk_recursive($data, function (&$item) {
+            if (is_string($item) && !mb_detect_encoding($item, 'utf-8', true)) {
```

Comment:
I just saw polyfill for 55 is already in composer.json, so please ignore that part of my comment.